client/my-sites/upgrades/domain-management/email-forwarding/email-forwarding-add-new.jsx

```diff
@@ -131,12 +157,14 @@ const EmailForwardingAddNew = React.createClass( {
 					<FormTextInputWithAffixes
 						disabled={ this.state.formSubmitting }
 						name="mailbox"
-						onChange={ this.handleFieldChange.bind( this, 'mailbox' ) }
+						onChange={ this.onChange }
 						onFocus={ this.handleFieldFocus.bind( this, 'Mailbox' ) }
+						isError={ ! isValidMailbox }
 						placeholder={ exampleEmailText }
 						type="text"
 						suffix={ '@' + this.props.selectedDomainName }
-						value={ this.state.mailbox } />
+						value={ mailbox } />
+					{ ! isValidMailbox ? <FormInputValidation text={ this.translate( 'Invalid mailbox' ) } isError={ true }/> : null }
```

Comment:
As discussed, maybe a more descriptive error could be used here - something like "Invalid mailbox - only XYZ characters are allowed". Otherwise, the user might be left scratching his/hers head as to why we are not allowing `<script type='text/javascript'>window.alert('Forwarding!')</script>` as the mailbox ;)